### PR TITLE
Updated dependencies + added hooks linting

### PR DESCRIPTION
**Pull request by InsaneViku**
Besides updating dependencies, the main things here were to:
- pop in `react/jsx-filename-extension': [WARNING, { 'extensions': ['.js'] }]`
- pop in `eslint-plugin-react-hooks`

### CODE REVIEW

File: react.js [hunk 1]
```diff
@@ -11,13 +11,22 @@ module.exports = {
...
     // Warn when a component should be specified as a stateless function if it doesn't require state/lifecycle methods.
     'react/prefer-stateless-function': OFF,
     // Warn when there's more than one expression per line in JSX.
     'react/jsx-one-expression-per-line': OFF,
+    // Allow .js files to have JSX
+    'react/jsx-filename-extension': [WARNING, { 'extensions': ['.js'] }],
+
+    /* React Hooks rules */
+    'react-hooks/rules-of-hooks': ERROR,
+    'react-hooks/exhaustive-deps': WARNING,
```
Comment (terryyu) [inline react.js]: there are few custom rules or exceptions i'd like to propose. they can be requested in another PR. looks good!

Comment (InsaneViku) [reply react.js]: yes please do! thanks!

**Comment by InsaneViku** [comment]
thanks @terryyu!
